slack/web/base_client.py

```diff
@@ -169,8 +174,18 @@ def api_call(
         if params is not None and isinstance(params, dict):
             params = {k: v for k, v in params.items() if v is not None}
 
+        token: Optional[str] = self.token
+        if params is not None and "token" in params:
+            token = params.pop("token")
+        if json is not None and "token" in json:
+            token = json.pop("token")
         req_args = {
-            "headers": self._get_headers(has_json, has_files, headers),
+            "headers": self._get_headers(
+                token=token,
```

Comment:
Is there any chance token is not defined by this step? Would that cause a problem?

Reply:
Tokens are not set for `oauth.access` and `oauth.v2.access` calls (req_args["auth"] is used for building Authorization header in a following step). I've verified this change doesn't break anything by manually testing with `integration_tests/samples/oauth/oauth_v2.py`.